tests/containers_test.go

```diff
@@ -13,49 +15,83 @@
 package containers
 
 import (
+	"encoding/json"
 	"fmt"
 	"os"
+	"os/exec"
 	"strings"
 	"testing"
 	"time"
 
 	"github.com/pulumi/pulumi/pkg/v3/testing/integration"
 	ptesting "github.com/pulumi/pulumi/sdk/v3/go/common/testing"
+	"github.com/stretchr/testify/require"
 )
 
-// TestPulumiDockerImage simulates building and running Pulumi programs on the pulumi/pulumi Docker image.
+type testCase struct {
+	template string
+	config   map[string]string
+}
+
+// TestPulumiTemplateTests simulates building and running Pulumi programs on the pulumi/pulumi Docker image.
 //
-// NOTE: This test is intended to be run inside the aforementioned container, unlike the actions test below.
-func TestPulumiDockerImage(t *testing.T) {
-	if os.Getenv("RUN_CONTAINER_TESTS") == "" {
-		t.Skip("Skipping container runtime tests because RUN_CONTAINER_TESTS not set.")
-	}
+// NOTE: This test is intended to be run inside the aforementioned container.
+func TestPulumiTemplateTests(t *testing.T) {
+	t.Parallel()
 
 	// Confirm we have credentials.
-	if os.Getenv("PULUMI_ACCESS_TOKEN") == "" {
-		t.Fatal("PULUMI_ACCESS_TOKEN not found, aborting tests.")
-	}
+	mustEnv(t, "PULUMI_ACCESS_TOKEN")
+	mustEnv(t, "ARM_CLIENT_ID")
+	mustEnv(t, "ARM_CLIENT_SECRET")
+	mustEnv(t, "ARM_TENANT_ID")
 
-	var stackOwner = os.Getenv("PULUMI_ORG")
-	if stackOwner == "" {
-		t.Fatal("PULUMI_ORG must be set.  Aborting tests.")
-	}
+	stackOwner := mustEnv(t, "PULUMI_ORG")
 
 	sdksToTest := []string{"csharp", "python", "typescript", "go", "java"}
 	if os.Getenv("SDKS_TO_TEST") != "" {
 		sdksToTest = strings.Split(os.Getenv("SDKS_TO_TEST"), ",")
 	}
+	clouds := []string{"azure" /*, "aws", "gcp"*/}
+	configs := map[string]map[string]string{
+		"azure": {
+			"azure-native:location": "EastUS",
+		},
+	}
+
+	testCases := []testCase{}
+	for _, sdk := range sdksToTest {
+		// python, typescript, ...
+		testCases = append(testCases, testCase{sdk, map[string]string{}})
+		if sdk == "java" {
```

Comment:
I don't want to change the containers in any way in this PR, so skipping some things that fail here, and below for azure-typescript.